### ISSUE TITLE
401 (unauthorized) error in iOS

### ISSUE THREAD

**Issue by allantatter** [ionic-team/ionic-view-issues#59]
I have created an app with ionic cli tools and uploaded it but when I want to sync with the latest version or refresh the app list on iOS in Ionic View app, then I get 401 error (on my iPhone 5 and my colleague's iPhone 6, with whom I have shared the app).

I have deleted the app and re-uploaded but there is no difference. On the other colleague's Android I get the updates and everything works as expected.

If this is important, then for uploading the app I run:

```
ionic build ios
ionic build android
ionic upload
```

![image1](https://cloud.githubusercontent.com/assets/386999/6842486/8b8f104c-d3a1-11e4-8067-1c696ef0beb6.PNG)
![image2](https://cloud.githubusercontent.com/assets/386999/6842487/8b91276a-d3a1-11e4-9b4b-92464577563a.PNG)

**Comment by tlancina** [comment]
Try logging out and logging back in

**Comment by allantatter** [comment]
Did that, at first it allowed to Sync to latest and said it was successful but actually it was soem old version from the past. After swiping with 3 fingers I got 401 errors again.

**Comment by tlancina** [comment]
Hmm... Does your app use local storage by chance?

**Comment by allantatter** [comment]
Yes it does. I store a few small strings there.

**Comment by tlancina** [comment]
Just trying to think what could cause your authentication to suddenly fail.
What are the names of your local storage keys?

**Comment by allantatter** [comment]
Key value pairs from my browser:

```
carModel	1	
carModelYear	1	
language	et_EE	
userHasAgreedTerms	1
```

**Comment by tlancina** [comment]
Thanks, that shouldn't cause any issues. If you don't clear local storage
anywhere either then I'm not really sure.

If you upload a blank ionic starter and load that (after re-logging back
in) do you have the same issue? Trying to figure out if it's caused by the
app you're viewing.

**Comment by allantatter** [comment]
In code I do following because I could not find other way to clear local storage without code:

```
$window.localStorage.clear();
```

Created blank starter app, uploaded.
Then downloaded via Ionic View, worked.
Made change in code, uploaded. Synced in Ionic view, worked and changes were there.
Synced the previous broken app, said it was successful but in reality was not the last version.
After swiping 3 fingers I got 401 errors with both apps.

**Comment by tlancina** [comment]
Ah ok, since you have only a few keys on localStorage, try just deleting them individually `$window.localStorage.removeItem(key)`.

Thanks for the issue, this is actually a fairly serious problem and should be fixed in the next release.

**Comment by allantatter** [comment]
I could change that in code, but I cannot update the app in Ionic View anyways :)

**Comment by tlancina** [comment]
That is most likely https://github.com/driftyco/ionic-view-issues/issues/10.

**Comment by allantatter** [comment]
Thanks, that was the other issue yes. Did not get 401 any more :)

tl;dr solution:

Replace this:

```
$window.localStorage.clear();
```

with this:

```
$window.localStorage.removeItem(key1)
$window.localStorage.removeItem(key2)
$window.localStorage.removeItem(key3)
```

**Comment by bjones6** [comment]
Just want to point out that modifying local storage in an ionic app still effects ionic view. Notably window.localStorage.clear() will eliminate your session in ionic view, causing the 401 error and a re-login for the user next time they open the app.

I just removed the call in favor of the suggested window.localStorage.removeItem(), but I imagine some ionic apps will need the .clear() method and it is rather annoying to have to login again afterwards.

Also Ionic is awesome.

**Comment by tlancina** [comment]
Hey thanks for the issue! With Ionic Pro we’re consolidating our support, so please head over to http://support.ionicjs.com if you’re having problems, thanks!
